pkg/kubernetes/cluster.go

```diff
@@ -0,0 +1,56 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package kubernetes
+
+import (
+	"fmt"
+	"io"
+	"io/ioutil"
+	"strings"
+)
+
+// ClusterDomain returns the k8s cluster domain extracted from the
+// /etc/resolv.conf.
+func ClusterDomain(resolvConf io.Reader) (string, error) {
+	data, err := ioutil.ReadAll(resolvConf)
+	if err != nil {
+		return "", fmt.Errorf("failed to get cluster domain: %w", err)
+	}
+
+	lines := strings.Split(string(data), "\n")
+	var searchLine string
+	for _, line := range lines {
+		if strings.HasPrefix(line, "search") {
+			searchLine = line
+		}
+	}
+
+	if searchLine == "" {
+		err := fmt.Errorf("missing the search path from resolv.conf")
+		return "", fmt.Errorf("failed to get cluster domain: %w", err)
+	}
+
+	domains := strings.Split(searchLine, " ")
+	for i := 1; i < len(domains); i++ {
```

Comment:
```suggestion
	for range _, search := range domains {
```
We'll needlessly look at `search`, but that doesn't start with `svc.`; if we use the external library, that's fine too.

Reply:
Err, this doesn't appear to have been fixed? By "if we use the external library" I just mean "we'll no longer have to worry about looking at the `search` word".

I'm not sure what happened to the "unresolve" button on GitHub.

Reply:
Oh dang, I forgot this part. Thanks for catching it!

Reply:
Fixed with 7262ee4.